modules/server/src/main/scala/higherkindness/compendium/storage/pg/implicits.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2018-2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.compendium.storage.pg
+
+import doobie.util.Meta
+import higherkindness.compendium.core.refinements.ProtocolId
+import higherkindness.compendium.models.Protocol
+
+object implicits {
+
+  implicit val protocolMeta: Meta[Protocol] =
+    Meta[Array[Byte]].timap[Protocol](binProto => Protocol(new String(binProto)))(_.raw.getBytes)
+
+  // TODO Blows up when retrieving an invalid protocol id
+  implicit val protocolIdMeta: Meta[ProtocolId] =
```

Comment:
Instead of use `Meta`, doobie encourages to define `Put` and `Get`. 

See https://github.com/tpolecat/doobie/blob/34a669c52948b2802bd4e028c0df43d1f3f3574b/modules/core/src/main/scala/doobie/util/get.scala#L58 to define a `Get` from a function that returns an `Either`. It could serve you so you do not need to use `_.right.get` :scream:

Reply:
Yeah, nice! Thanks for the tip!